# Python Project (Netflix Database)

In [50]:
import pandas as pd

In [51]:
df = pd.read_csv('titles_g.csv')

In [52]:
type(df)

pandas.core.frame.DataFrame

## Q1.) What are the most well recieved genres on Netflix based on IMDB and TMDB score
####    

In [202]:
df.head(10)

id                                title   type  \
0  ts300399  Five Came Back: The Reference Films   SHOW   
1   tm84618                          Taxi Driver  MOVIE   
2  tm154986                          Deliverance  MOVIE   
3  tm127384      Monty Python and the Holy Grail  MOVIE   
4  tm120801                      The Dirty Dozen  MOVIE   
5   ts22164         Monty Python's Flying Circus   SHOW   
6   tm70993                        Life of Brian  MOVIE   
7   tm14873                          Dirty Harry  MOVIE   
8  tm119281                     Bonnie and Clyde  MOVIE   
9   tm98978                      The Blue Lagoon  MOVIE   

                                         description  release_year  \
0  This collection includes 12 World War II-era p...          1945   
1  A mentally unstable Vietnam War veteran works ...          1976   
2  Intent on seeing the Cahulawassee River before...          1972   
3  King Arthur, accompanied by his squire, recrui...          1975   
4  12 American military prisoners in World War II...          1967   
5  A British sketch comedy series with the shows ...          1969   
6  Brian Cohen is an average young Jewish man, bu...          1979   
7  When a madman dubbed 'Scorpio' terrorizes San ...          1971   
8  In the 1930s, bored waitress Bonnie Parker fal...          1967   
9  Two small children and a ship's cook survive a...          1980   

  age_certification  runtime             genres production_countries  seasons  \
0             TV-MA       51  ['documentation']               ['US']      1.0   
1                 R      114          ['drama']               ['US']      NaN   
2                 R      109          ['drama']               ['US']      NaN   
3                PG       91        ['fantasy']               ['GB']      NaN   
4               NaN      150            ['war']         ['GB', 'US']      NaN   
5             TV-14       30         ['comedy']               ['GB']      4.0   
6                 R       94         ['comedy']               ['GB']      NaN   
7                 R      102       ['thriller']               ['US']      NaN   
8                 R      110          ['crime']               ['US']      NaN   
9                 R      104        ['romance']               ['US']      NaN   

     imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  genreNum  
0        NaN         NaN         NaN            0.600         NaN       NaN  
1  tt0075314         8.2    808582.0           40.965       8.179       NaN  
2  tt0068473         7.7    107673.0           10.010       7.300       NaN  
3  tt0071853         8.2    534486.0           15.461       7.811       NaN  
4  tt0061578         7.7     72662.0           20.398       7.600       NaN  
5  tt0063929         8.8     73424.0           17.617       8.306       NaN  
6  tt0079470         8.0    395024.0           17.770       7.800       NaN  
7  tt0066999         7.7    155051.0           12.817       7.500       NaN  
8  tt0061418         7.7    112048.0           15.687       7.500       NaN  
9  tt0080453         5.8     69844.0           50.324       6.156       NaN

In [54]:
df.size

87750

### The columns within the data frame are

In [55]:
df.axes

[RangeIndex(start=0, stop=5850, step=1),
 Index(['id', 'title', 'type', 'description', 'release_year',
        'age_certification', 'runtime', 'genres', 'production_countries',
        'seasons', 'imdb_id', 'imdb_score', 'imdb_votes', 'tmdb_popularity',
        'tmdb_score'],
       dtype='object')]

In [201]:
df[['title','genres', 'type','imdb_score', 'tmdb_score']].head(10)

title             genres   type  imdb_score  \
0  Five Came Back: The Reference Films  ['documentation']   SHOW         NaN   
1                          Taxi Driver          ['drama']  MOVIE         8.2   
2                          Deliverance          ['drama']  MOVIE         7.7   
3      Monty Python and the Holy Grail        ['fantasy']  MOVIE         8.2   
4                      The Dirty Dozen            ['war']  MOVIE         7.7   
5         Monty Python's Flying Circus         ['comedy']   SHOW         8.8   
6                        Life of Brian         ['comedy']  MOVIE         8.0   
7                          Dirty Harry       ['thriller']  MOVIE         7.7   
8                     Bonnie and Clyde          ['crime']  MOVIE         7.7   
9                      The Blue Lagoon        ['romance']  MOVIE         5.8   

   tmdb_score  
0         NaN  
1       8.179  
2       7.300  
3       7.811  
4       7.600  
5       8.306  
6       7.800  
7       7.500  
8       7.500  
9       6.156

In [67]:
grouped = df.groupby('genres')

In [68]:
print (grouped)

## Grouping genres

In [136]:
for name, data in grouped:
    print(name)

['action']
['animation']
['comedy']
['crime']
['documentation']
['drama']
['family']
['fantasy']
['history']
['horror']
['music']
['other']
['reality']
['romance']
['scifi']
['sport']
['thriller']
['war']
['western']


In [175]:
df.groupby(['genres','imdb_score','tmdb_score'], as_index=False).mean().groupby('genres')[['imdb_score','tmdb_score']].mean()

imdb_score  tmdb_score
genres                                   
['action']           6.284722    6.748455
['animation']        6.513393    7.358268
['comedy']           6.215979    6.575818
['crime']            6.732778    6.938144
['documentation']    6.986911    7.132552
['drama']            6.657781    6.923018
['family']           6.200000    7.072927
['fantasy']          6.196000    6.707560
['history']          6.685714    6.935524
['horror']           5.431915    5.883777
['music']            6.714286    7.106333
['other']            7.050000    6.900000
['reality']          6.254422    7.149660
['romance']          6.038947    6.417379
['scifi']            6.687685    7.175054
['sport']            6.675000    6.800000
['thriller']         6.138144    6.436921
['war']              6.979070    6.838651
['western']          6.233333    6.299867

### Genres and their abbreviations
    1. action 'ac'            
    2. animation 'an'         
    3. comedy 'co'   
    4. crime 'cr' 
    5. documentation 'doc'
    6. drama 'dr'
    7. family 'fam'
    8. fantasy 'fa'
    9. history 'hi'
    10. horror 'ho'
    11. music 'mu'
    12. other 'ot'
    13. reality 're'
    14. romance 'ro'
    15. scifi 'sci'
    16. sport 'sp'
    17. thriller 'th'
    18. war 'war'
    19. western 'wes'

In [199]:
ac = (6.284722 + 6.748455)/2
an = (6.513393 + 7.358268)/2
co = (6.215979 + 6.575818)/2
cr = (6.732778 + 6.938144)/2
doc = (6.986911 + 7.132552)/2
dr = (6.657781 + 6.923018)/2
fam = (6.200000 + 7.072927)/2
fa = (6.196000 + 6.707560)/2
hi = (6.685714 + 6.935524)/2
ho = (5.431915 + 5.883777)/2
mu = (6.714286 + 7.106333)/2
ot = (7.050000 + 6.900000)/2
re = (6.254422 + 7.149660)/2
ro = (6.038947 + 6.417379)/2
sci = (6.687685 + 7.175054)/2
sp = (6.675000 + 6.800000)/2
th = (6.138144 + 6.436921)/2
war = (6.979070 + 6.838651)/2
wes = (6.233333 + 6.299867)/2
print("Total average of IMDB and TMDB scores")
print("documentation: " + str(doc))
print("other: " + str(ot))
print("animation: " + str(an))
print("scifi: " + str(sci))
print("music: " + str(mu))
print("war: " + str(war))
print("crime: " + str(cr))
print("history: " + str(hi))
print("drama: " + str(dr))
print("sport: " + str(sp))
print("reality: " + str(re))
print("family: " + str(fam))
print("action: " + str(ac))
print("fantasy: " + str(fa))
print("comedy: " + str(co))
print("thriller: " + str(th))
print("western: " + str(wes))
print("romance: " + str(ro))
print("horror: " + str(ho))

Total average of IMDB and TMDB scores
documentation: 7.0597315
other: 6.975
animation: 6.9358305
scifi: 6.931369500000001
music: 6.9103095
war: 6.908860499999999
crime: 6.8354610000000005
history: 6.810619
drama: 6.7903994999999995
sport: 6.7375
reality: 6.7020409999999995
family: 6.6364635
action: 6.5165885
fantasy: 6.451779999999999
comedy: 6.3958984999999995
thriller: 6.287532499999999
western: 6.2666
romance: 6.228163
horror: 5.657846


### Top 5 Genres based off of IMDB and TMDB score
            1. Documentation: 7.059
            2. Other: 6.975
            3. Animation: 6.935
            4. Scifi: 6.931
            5. Music: 6.910

## Q2.) Does the runtime of a show or movie affect the popularity?

In [203]:
df[['title', 'runtime','imdb_votes', 'tmdb_popularity']].head(10)

title  runtime  imdb_votes  tmdb_popularity
0  Five Came Back: The Reference Films       51         NaN            0.600
1                          Taxi Driver      114    808582.0           40.965
2                          Deliverance      109    107673.0           10.010
3      Monty Python and the Holy Grail       91    534486.0           15.461
4                      The Dirty Dozen      150     72662.0           20.398
5         Monty Python's Flying Circus       30     73424.0           17.617
6                        Life of Brian       94    395024.0           17.770
7                          Dirty Harry      102    155051.0           12.817
8                     Bonnie and Clyde      110    112048.0           15.687
9                      The Blue Lagoon      104     69844.0           50.324

In [205]:
grouped = df.groupby('runtime')

In [206]:
for name, data in grouped:
    print(name)

0
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
184
185
186
187
188
189
190
191
192
194
196
200
206
208
209
210
214
224
225
240


In [216]:
df.groupby(['runtime', 'tmdb_popularity'], as_index=False).mean().groupby('runtime')['tmdb_popularity'].mean().head(50)

runtime
0      3.759357
2      5.909167
3      3.173429
4      3.476500
5      4.720667
6      8.163143
7     10.468800
8      4.974600
9      9.370700
10    12.531385
11    29.837417
12    10.744905
13    16.717208
14    11.131467
15    12.157043
16    10.121636
17    12.939800
18    17.196125
19    14.967000
20     7.882150
21    21.897757
22    19.338983
23    21.317456
24    33.161320
25    25.922459
26    15.765792
27    14.359902
28    15.486405
29    10.528887
30    10.067650
31    25.893192
32     8.458964
33    22.979276
34    13.321609
35     5.183609
36    12.930591
37     4.895050
38    45.823059
39    10.207143
40    13.702325
41    30.877425
42    43.351389
43    29.166813
44    25.201370
45    29.539156
46    26.560453
47    41.846525
48    22.151456
49    34.836893
50    25.261492
Name: tmdb_popularity, dtype: float64

In [217]:
df.groupby(['runtime', 'tmdb_popularity'], as_index=False).mean().groupby('runtime')['tmdb_popularity'].mean().tail(50)

runtime
153      4.555800
154      4.913000
155      5.229583
156      7.202125
157      7.460375
158     29.029500
159      3.872750
160      4.962857
161      4.181444
162      6.751000
163     13.955667
164     25.018333
165     25.330250
166     10.034000
167      8.740000
168     38.877000
169      6.329333
170     10.839333
171      4.808667
172      5.047200
173      6.211500
174      3.328500
175     24.974667
176      7.084000
177      7.865000
178      2.626000
179     15.036333
180      6.447500
181      1.542000
182      1.121500
184      7.562500
185     12.350000
186      7.818667
187     53.526000
188     26.531000
189      9.019000
190      1.487000
191     14.677000
192      1.912000
194    155.683000
196      4.147341
200      8.529000
206      6.314000
208      7.580000
209     21.075000
210     13.699000
214     11.959000
224     10.097000
225      2.821000
240      0.933000
Name: tmdb_popularity, dtype: float64

### There is no conclusive evidence when it comes to whether or not the runtime of a show affects its popularity